File: backend/src/main/java/com/orange/Crisalis/model/dto/OrderDTO.java
package com.orange.Crisalis.model.dto;

import com.orange.Crisalis.enums.OrderState;
import com.orange.Crisalis.model.ClientEntity;

import com.orange.Crisalis.model.OrderEntity;
import com.orange.Crisalis.model.SellableGood;
import lombok.*;

import java.io.Serializable;
import java.util.Date;
import java.util.List;
import java.util.stream.Collectors;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class OrderDTO implements Serializable {
    private Long id;
    private Date dateCreated;
    private OrderState orderState;
    private ClientEntity client;
    private SellableGood service;
    private List<OrderDetailDTO> orderDetailDTOList;

    public OrderDTO(OrderEntity order) {
        this.id = order.getId();
        this.client = order.getClient();
        this.orderState = order.getOrderState();
        this.dateCreated = order.getDateCreated();
        this.orderDetailDTOList = order.getOrderDetailList().stream().map(OrderDetailDTO::new).collect(Collectors.toList());
        this.service = order.getService();
    }

}

File: frontend/src/app/order/model/order-dto.ts
import { OrderState } from "./order-state";
import { ClientEntity } from "./client-entity";
import { OrderDetailDTO } from "./order-details-dto";
import { SellableGood } from "src/app/modules/sellable-good/model/sellable-good.model";

export class OrderDTO {
  id?: number;
  client: ClientEntity;
  orderState: OrderState;
  dateCreated: Date;
  orderDetailDTOList: OrderDetailDTO[];
  service?:SellableGood;

  constructor(id: number, client: ClientEntity, orderState: OrderState, dateCreated: Date, orderDetailDTOList: OrderDetailDTO[]) {
    this.id = id;
    this.client = client;
    this.orderState = orderState;
    this.dateCreated = dateCreated;
    this.orderDetailDTOList = orderDetailDTOList;
  }
}
